### PR TITLE
Add app baseball walk scoring

### LINKED ISSUE #1572: Implement Walk Action in Baseball Live Scoring for New App
## Signal Feature
Enable scorekeepers to record a Walk during a live baseball game, updating bases, scores, and pitch count.

## Functional Gap
The legacy HTML/JS app can apply a Walk action in live baseball scoring, automatically advancing runners and updating scores, which the new React/native app cannot currently do.

## Missing Signal Delta
The specific state transition logic for recording a Walk (resetting balls/strikes, advancing runners based on occupied bases, and incrementing runs) is absent in the new app.

## Verification
**Priority:** high
**Confidence:** high
**Focus area:** live-scoring

### Existing New-App Capability
- `apps/app/src/pages/GameDetail.tsx: This component likely displays general game information and scores, serving as the main interface for a live game.`
- `apps/app/src/lib/gameReportService.ts: This service might handle basic game report generation or score aggregation, but not live, sport-specific state transitions.`

### Absence Verification
- `grep -r "applyWalk" /home/paul-bot1/.local/state/paul-bot1/mobile-app-parity-gap-researcher/workspaces/pauljsnider__allplays/apps/app/src (returned no results)`
- `grep -rE "baseball|inning|bases|balls|strikes|outs|homeRun|single|double|triple|walk" /home/paul-bot1/.local/state/paul-bot1/mobile-app-parity-gap-researcher/workspaces/pauljsnider__allplays/apps/app/src/pages/GameDetail.tsx /home/paul-bot1/.local/state/paul-bot1/mobile-app-parity-gap-researcher/workspaces/pauljsnider__allplays/apps/app/src/lib/gameReportService.ts (returned no results)`

### Legacy Web Logic
- `/home/paul-bot1/.local/state/paul-bot1/mobile-app-parity-gap-researcher/workspaces/pauljsnider__allplays/js/live-scorekeeping-baseball.js:applyWalk(state): This function contains the core logic for advancing runners and scoring runs when a walk occurs.`
- `/home/paul-bot1/.local/state/paul-bot1/mobile-app-parity-gap-researcher/workspaces/pauljsnider__allplays/js/live-scorekeeping-baseball.js:addRuns(state, runs): Utility function to add runs to the appropriate team's score.`
- `/home/paul-bot1/.local/state/paul-bot1/mobile-app-parity-gap-researcher/workspaces/pauljsnider__allplays/js/live-scorekeeping-baseball.js:resetCount(state): Resets balls and strikes after a play like a walk.`

### New iOS / Android / React App State
- `apps/app/src/pages/GameDetail.tsx: This component lacks the UI elements and event handlers to trigger a 'Walk' action with baseball-specific logic.`
- `apps/app/src/lib/gameReportService.ts: This service does not appear to contain the sport-specific state management functions for live baseball scoring actions.`

## Technical Architecture
A new service/module within `apps/app/src/lib/sportScoring` would encapsulate the baseball-specific state mutations for actions like Walk. The `GameDetail.tsx` component would integrate with this new service, providing a UI button to trigger the Walk action and displaying the updated game state. Native wrappers would render the updated React component.

## UX Requirements
- A clear, tappable 'Walk' button is visible to the scorekeeper during a live baseball game.
- Upon tapping 'Walk', the UI instantly reflects the updated base runners, score, and pitch count (balls/strikes reset).
- The UI must visually represent occupied bases accurately after a walk event.

## Requirements
- The system must correctly advance runners from first, second, and third base upon a walk, including scoring runs if bases are loaded.
- The system must reset the ball/strike count to 0-0 after a walk.
- The game state (inning, half, outs, scores) must be updated correctly after applying a walk.

## Implementation
- Create a new `baseballScorekeepingService.ts` within `apps/app/src/lib/sportScoring` to house the `applyWalk` and related baseball state logic.
- Integrate the `baseballScorekeepingService` into `apps/app/src/pages/GameDetail.tsx`.
- Add a 'Walk' button to the `GameDetail.tsx` UI that calls the new service's `applyWalk` function.
- Update the UI in `GameDetail.tsx` to display the new bases, score, and pitch count after a walk.

## Non-Goals
- Implementing all baseball live-scoring actions (e.g., strikeout, single, double, triple, home run, out).
- Refactoring the entire `GameDetail.tsx` component or the broader game state management system.
- Adding an undo/redo mechanism for all scoring actions.

### Unit
- Test `applyWalk` function: given loaded bases, verify 1 run scored and bases remain loaded (first, second, third occupied).
- Test `applyWalk` function: given a runner on first, verify runners on first and second, no runs scored.
- Test `applyWalk` function: verify balls and strikes are reset to zero after a walk.

### Integration
- Simulate a 'Walk' action via the UI in `GameDetail.tsx` and verify the displayed game state (score, bases, pitch count) updates correctly.
- Verify that the 'Walk' action successfully persists the updated game state to the backend (mocked or actual API call).

### Functional
- Manually navigate to a live baseball game in the React web app and tap the 'Walk' button. Verify bases advance, score updates, and pitch count resets correctly on screen.
- Run Playwright test: navigate to live baseball scoring, trigger 'Walk' action, assert visible state changes match expected baseball rules.

## Likely Files
- `apps/app/src/pages/GameDetail.tsx`
- `apps/app/src/lib/sportScoring/baseballScorekeepingService.ts`

## Deduplication Notes
Searched existing file names and content for explicit baseball scoring actions like walk or bases in state management contexts within the new React app. No direct matches or obvious in-progress PRs for this specific signal feature.

---
Detected by `paulbot-mobile-app-parity-gap-researcher` automated hourly scan.

### PR DESCRIPTION
**Pull request by pauljsnider**
Closes #1572

Recovered from a partial issue-fixer run. The agent committed the fix branch successfully, so paulbot is opening the PR from the recorded commit.

### CODE REVIEW

File: apps/app/src/lib/sportScoring/baseballScorekeepingService.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+export function applyWalk(state: Partial<BaseballLiveState>): BaseballScoringResult {
+  const normalized = createBaseballLiveState(state);
+  const { bases } = normalized;
+  const runs = bases.first && bases.second && bases.third ? 1 : 0;
+  const nextBases: BaseballBases = {
+    first: true,
+    second: bases.first || bases.second,
+    third: bases.third || (bases.first && bases.second)
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply apps/app/src/lib/sportScoring/baseballScorekeepingService.ts:86]: ### Remediation Update

Addressed this review feedback pass on PR #1627.

- Commit: `ba2389a161017d2dc7258b215dcb88baa3ef111f`
- Validation: pass: 

{"success":true,"files_changed":["apps/app/src/pages/GameDetail.tsx","tests/unit/app-baseball-scorekeeping-service.test.ts","tests/unit/app-game-detail-baseball-walk.test.jsx"],"commit_message":"Fix baseball scoring review findings","threads_addressed":["PRRT_kwDOQe-T586F8xMd","PRRT_kwDOQe-T586F8xMg","PRRT_kwDOQe-T586F8xV-","PRRT_kwDOQe-T586F8xV_"],"summary":"I fixed GameDetail so its hooks always run before the missing-game redirect and so baseball live scoring only renders for coach/admin users on baseball or softball teams. I also added focused regression tests for baseball-only rendering, redirect stability, and the walk edge cases, which confirmed the existing top-half score assertion and runner-on-third walk behavior were already correct."}

File: tests/unit/app-game-detail-baseball-walk.test.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        });
+        await act(async () => {
+            container.querySelector('[data-testid="baseball-base-third"]').click();
+        });
+        await act(async () => {
+            getButton(container, 'Walk').click();
+        });
+
+        expect(container.textContent).toContain('Walk, 1 run scored');
+        expect(container.textContent).toContain('Bears 0 · Falcons 1');
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply tests/unit/app-game-detail-baseball-walk.test.jsx]: ### Remediation Update

Addressed this review feedback pass on PR #1627.

- Commit: `ba2389a161017d2dc7258b215dcb88baa3ef111f`
- Validation: pass: 

{"success":true,"files_changed":["apps/app/src/pages/GameDetail.tsx","tests/unit/app-baseball-scorekeeping-service.test.ts","tests/unit/app-game-detail-baseball-walk.test.jsx"],"commit_message":"Fix baseball scoring review findings","threads_addressed":["PRRT_kwDOQe-T586F8xMd","PRRT_kwDOQe-T586F8xMg","PRRT_kwDOQe-T586F8xV-","PRRT_kwDOQe-T586F8xV_"],"summary":"I fixed GameDetail so its hooks always run before the missing-game redirect and so baseball live scoring only renders for coach/admin users on baseball or softball teams. I also added focused regression tests for baseball-only rendering, redirect stability, and the walk edge cases, which confirmed the existing top-half score assertion and runner-on-third walk behavior were already correct."}

File: apps/app/src/pages/GameDetail.tsx [hunk 3]
```diff
@@ -12,6 +14,25 @@ export function GameDetail({ auth }: { auth: AuthState }) {
   }
 
   const players = mockPlayers.filter((player) => game.playerIds.includes(player.id));
+  const [baseballState, setBaseballState] = useState<BaseballLiveState>(() => createBaseballLiveState());
+  const [lastScoringAction, setLastScoringAction] = useState('Ready for pitch');
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply apps/app/src/pages/GameDetail.tsx]: ### Remediation Update

Addressed this review feedback pass on PR #1627.

- Commit: `ba2389a161017d2dc7258b215dcb88baa3ef111f`
- Validation: pass: 

{"success":true,"files_changed":["apps/app/src/pages/GameDetail.tsx","tests/unit/app-baseball-scorekeeping-service.test.ts","tests/unit/app-game-detail-baseball-walk.test.jsx"],"commit_message":"Fix baseball scoring review findings","threads_addressed":["PRRT_kwDOQe-T586F8xMd","PRRT_kwDOQe-T586F8xMg","PRRT_kwDOQe-T586F8xV-","PRRT_kwDOQe-T586F8xV_"],"summary":"I fixed GameDetail so its hooks always run before the missing-game redirect and so baseball live scoring only renders for coach/admin users on baseball or softball teams. I also added focused regression tests for baseball-only rendering, redirect stability, and the walk edge cases, which confirmed the existing top-half score assertion and runner-on-third walk behavior were already correct."}

File: apps/app/src/pages/GameDetail.tsx [hunk 4]
```diff
@@ -12,6 +14,25 @@ export function GameDetail({ auth }: { auth: AuthState }) {
...
+  const canScoreBaseball = auth.isCoach || auth.isAdmin;
```
[the remark this answers is not in the document]
Comment (pauljsnider) [reply apps/app/src/pages/GameDetail.tsx]: ### Remediation Update

Addressed this review feedback pass on PR #1627.

- Commit: `ba2389a161017d2dc7258b215dcb88baa3ef111f`
- Validation: pass: 

{"success":true,"files_changed":["apps/app/src/pages/GameDetail.tsx","tests/unit/app-baseball-scorekeeping-service.test.ts","tests/unit/app-game-detail-baseball-walk.test.jsx"],"commit_message":"Fix baseball scoring review findings","threads_addressed":["PRRT_kwDOQe-T586F8xMd","PRRT_kwDOQe-T586F8xMg","PRRT_kwDOQe-T586F8xV-","PRRT_kwDOQe-T586F8xV_"],"summary":"I fixed GameDetail so its hooks always run before the missing-game redirect and so baseball live scoring only renders for coach/admin users on baseball or softball teams. I also added focused regression tests for baseball-only rendering, redirect stability, and the walk edge cases, which confirmed the existing top-half score assertion and runner-on-third walk behavior were already correct."}
